templates/ts-scrape-single-page/main.ts

```diff
@@ -0,0 +1,40 @@
+// Axios - Promise based HTTP client for the browser and node.js (Read more at https://axios-http.com/docs/intro).
+import axios from "axios";
+// Cheerio - The fast, flexible & elegant library for parsing and manipulating HTML and XML (Read more at https://cheerio.js.org/).
+import * as cheerio from "cheerio";
+// Apify SDK - toolkit for building Apify Actors (Read more at https://docs.apify.com/sdk/js/).
+import { Actor } from "apify";
+
+// The init() call configures the Actor for its environment. It's recommended to start every Actor with an init().
+await Actor.init();
+
+interface Input {
+    url: string;
+}
+// Structure of input is defined in input_schema.json
+const input = await Actor.getInput<Input>();
+if (!input) throw new Error("Input is missing!");
```

Comment:
@B4nan Is there a better way to go around this than throwing an Error? This works as well
```
const input = await Actor.getInput() as Input;
```
But I am not sure which one is better.

Reply:
I think we should have a method that throws when the input is not there in the actor sdk, it's quite common practice, but since we don't have one, I would keep it the way it is now, with the explicit throw. Better than lying to the compiler (which you would do with the type assertion).

I can add `Actor.getInputOrThrow()` there if you want, it makes total sense to me (actually I would be fine throwing from the `getInput` directly, but that's quite breaking).

Reply:
Adding the `getInputOrThrow` would probably require a lot of udpate in documentation as well right? I think for now I will just throw the error :) thx

Reply:
Not really, just a new method, with a jsdoc, we don't need to update existing docs because of that (at least not immediately). I will add that anyway, I wanted to do that for quite some time.

Reply:
But we don't need the throw if we type the input, no? If we say that the input is always an object with a url: string, then we can delete the line with the exception.

I'd say that's a better tutorial pattern for a TS template than validation. Tutorials and examples usually skip validation, because if you validate one thing, then you should validate everything, add error handling and what not, so it's typical to just follow the happy path in tutorials to keep them clean and succinct.

Reply:
> But we don't need the throw if we type the input, no? If we say that the input is always an object with a url: string, then we can delete the line with the exception.

But it is not always an object, if the INPUT key is missing from the KVS, it returns `null`. And that is what we want to deal with. I guess your point of view is that the INPUT key will be part of the template, so it will be always there? Still, users can wipe the storage manually, and their code would start failing with some reference error etc, throwing is the cleaner approach.

So yes, we can assert the type, but it's still lying to the compiler, unlike conditional throwing, which is a runtime statement (that propagates to the type level).

Reply:
Yeah, I know how it works and you also got my point correctly.

Tutorials are always "incomplete". You never account for all the possible scenarios, because that's not the point. The point is to showcase functionality in the smallest possible example that includes only what it needs to and nothing else.

Plus, even the deletion of storage directory is only a temporary problem, because after we support input schema prefills / defaults in CLI, the URL will always have a value.

But I don't have a strong opinion about it, I just think it doesn't look good and adds unnecessary noise. I never put validation in any tutorials / examples, unless it actually showcases something I'm writing about.